json_object.h

```diff
@@ -458,29 +597,104 @@ JSON_EXPORT json_bool json_object_object_get_ex(const struct json_object *obj, c
 JSON_EXPORT void json_object_object_del(struct json_object *obj, const char *key);
 
 /**
- * Iterate through all keys and values of an object.
+ * @brief Delete the given json_object field
+ *
+ * The reference count will be decremented for the deleted object.  If there
+ * are no more owners of the value represented by this key, then the value is
+ * freed.  Otherwise, the reference to the value will remain in memory.
+ *
+ * @param obj the json_object instance
+ * @param key the object field name,
+ *            which is not terminated by a NULL ( @c '\0' ) character
+ * @param len the length of @p key
+ */
+JSON_EXPORT void json_object_object_del_len(struct json_object *jso, const char *key,
+                                            const size_t len);
+
+/**
+ * @brief Delete the given json_object field
+ *
+ * The reference count will be decremented for the deleted object.  If there
+ * are no more owners of the value represented by this key, then the value is
+ * freed.  Otherwise, the reference to the value will remain in memory.
+ *
+ * @param obj the json_object instance
+ * @param key the object field name
+ */
+JSON_EXPORT int json_object_object_del_key(struct json_object *obj, const struct json_key *key);
+
+/**
+ * @brief Iterate through all keys and values of an object.
+ *
+ * Adding keys to the object while iterating is NOT allowed.
+ *
+ * Deleting an existing key, or replacing an existing key with a
+ * new value IS allowed.
+ *
+ * @param obj the json_object instance
+ * @param key the local name for the `char *` key variable defined in the body
+ * @param val the local name for the `json_object *` object variable defined in
+ *            the body
+ */
+#if defined(__GNUC__) && !defined(__STRICT_ANSI__) && \
+    (defined(__STDC_VERSION__) && __STDC_VERSION__ >= 199901L)
+
+#define json_object_object_foreach(obj, key, val)                                            \
```

Comment:
The json_object_object_foreach_len() macro duplicates most of the code for json_object_object_foreach().  How about instead of doing that we have json_object_object_foreach() define a variable in addition to "key", perhaps "key_with_len" (though I'm not too excited about that var name)?